shared/src/native-src/loader.cpp

```diff
@@ -667,8 +671,8 @@ namespace shared
             return S_FALSE;
         }
 
-        bool disableNGenForFunction = (0 == memcmp(functionName, SpecificMethodToInjectName, sizeof(WCHAR) * (std::min)(NameBuffSize, functionNameLength)))
-                                        && (0 == memcmp(typeName, SpecificTypeToInjectName, sizeof(WCHAR) * (std::min)(NameBuffSize, typeNameLength)));
+        bool disableNGenForFunction = (0 == memcmp(functionName, SpecificMethodToInjectName, sizeof(WCHAR) * std::min( { SpecificMethodToInjectNameSize, NameBuffSize, (size_t)functionNameLength } )))
```

Comment:
For better readability create a function that:
1. test for size equality (i.e. SpecificMethodToInjectNameSize == functionName_Length_(Size) and SpecificTypeToInjectNameSize == typeName_Length_(Size)
2. test for content (the current code change looks good to me even though "foo" and "foo1" would be seen as "equal" but not an issue in our context)